### PR TITLE
Manual escape room config

### PR DESCRIPTION
**Pull request by robinjo78**
Allows the user to manually fill in the configuration of the escape room in the application.

### CODE REVIEW

**Comment by lisv** [comment]
I would use captital letters for the menuOption "Add extra media" -> "Add Extra Media" (to match the other menu options), or just "Add Media" would maybe be better

**Comment by lisv** [comment]
I would like the pop-up menu to be bigger, the rest of the screen does not matter when configuring. So I would make it wider, and the text bigger, as it is now quite small and not so userfriendly

**Comment by lisv** [comment]
Also, it is not clear what target duration in sec means, because it means the target duration for that single chest. Could there be a way to make this more clear?

File: src/main/java/gui/panes/ManualConfigPane.java [hunk 1]
```diff
@@ -0,0 +1,407 @@
...
+    /**
+     * Creates the manualConfig stage and panes.
+     * @param manual the menuItem
+     */
+    public void createManualConfig(final MenuItem manual, final Stage primaryStage) {
+        manual.setOnAction(t -> {
+            if (videoController.isClosed()) {
+                manualStage.setTitle("Manual Escape Room Configuration");
+//                manualStage.initModality(Modality.APPLICATION_MODAL);
+//                manualStage.initOwner(primaryStage);
```
Comment (egedebruin) [inline src/main/java/gui/panes/ManualConfigPane.java]: ahum...

File: src/main/java/gui/panes/ManualConfigPane.java [hunk 2]
```diff
@@ -0,0 +1,407 @@
...
+    private void createChestsLine(final TextField chestField) {
+        final Label chests = new Label("Amount of chests: ");
+        chests.setFont(font);
+
+        chestField.setMaxWidth(maxWidth);
+
+        fillInPane.add(chests, 0, 1);
+        fillInPane.add(chestField, 1, 1);
+    }
+
```
Comment (egedebruin) [inline src/main/java/gui/panes/ManualConfigPane.java]: createPlayersLine and createChestsLine seem to be very similar methods, isn't there a ways to improve this

File: src/main/java/gui/controllers/RoomController.java [hunk 3]
```diff
@@ -32,12 +32,18 @@
...
+     * @param configFile the configFile for this room.
      */
     public void configure(final String configFile) {
         progress = new Progress(configFile);
+        configure();
+    }
 
+    /**
+     * Configures everything needed.
+     */
```
Comment (Jasperkroes) [inline src/main/java/gui/controllers/RoomController.java]: en dat is????

File: src/main/java/gui/panes/ManualConfigPane.java [hunk 4]
```diff
@@ -0,0 +1,407 @@
...
+import javafx.stage.Stage;
+
+import java.util.ArrayList;
+
+/**
+ * Class that creates the ManualConfigPane in a new Scene.
+ */
+public class ManualConfigPane {
+    /**
+     * Class parameters.
```
Comment (Jasperkroes) [inline src/main/java/gui/panes/ManualConfigPane.java]: dit hoeft er echt niet bij hoor, dat is wel duidelijk

**Comment by Jasperkroes** [comment]
image

Dit krijg ik als ik de config voor de 2e keer open

**Comment by egedebruin** [comment]
When I add a camera or stream manually the amount of cameras stay 0.

**Comment by lisv** [comment]
* I don't think it is nice to fill in the duration of the escape in seconds. Minutes would be better. Because when does it ever come down to seconds? "For this escape room you have 25 minutes en 47 seconds" ? Not very nice.
* One should not be able to change the amount of people, chests, and duration time at the second stage of the pop-up, since this is only confusing (as is does not have any effect on the rest of the settings).
* The Proceed/Submit buttons are confusing.. On which one should I click?? Oh I get it now, still maybe it would be better to rename "Proceed" to "Adjust Settings" to make it more clear.
* The third menu option is called Manual Configuration, just like the option inside the second menu option. This should be renamed to "Add media" as it was before.

I do like that this manual configuration allows one to use the full interface (progressbar, status, warningpane etc.) with regular video files (instead of streams).

File: src/main/java/gui/panes/ManualConfigPane.java [hunk 5]
```diff
@@ -0,0 +1,370 @@
...
+     * @param vControl the videoController
+     * @param rControl the roomController
+     */
+    public ManualConfigPane(final VideoController vControl, final RoomController rControl) {
+        this.videoController = vControl;
+        this.roomController = rControl;
+        this.manualStage = new Stage();
+        this.fillInPane = new GridPane();
+
+        final int fontSize = 20;
```
Comment (lisv) [inline src/main/java/gui/panes/ManualConfigPane.java]: Font and font size can also be placed in the css file and then add a styleclass to that specific pane. May be better for separation of functionalities. But not a super big deal

File: src/main/java/gui/panes/ManualConfigPane.java (hunk 2 above)
Comment (lisv) [reply src/main/java/gui/panes/ManualConfigPane.java]: Yes, maybe give a boolean or string to the method to differentiate between the two strings and place within the pane.

**Comment by robinjo78** [comment]
@lisv @Jasperkroes de naam van het derde menu is weer veranderd.

**Comment by Jasperkroes** [comment]
> The Proceed/Submit buttons are confusing.. On which one should I click?? Oh I get it now, still maybe it would be better to rename "Proceed" to "Adjust Settings" to make it more clear.

Is dit al gefixt?

**Comment by robinjo78** [comment]
@Jasperkroes  JA!

**Pull request merged**